plugins/inputs/prometheus/kubernetes.go

```diff
@@ -0,0 +1,161 @@
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/ericchiang/k8s"
+	corev1 "github.com/ericchiang/k8s/apis/core/v1"
+	"github.com/ghodss/yaml"
+)
+
+// loadClient parses a kubeconfig from a file and returns a Kubernetes
+// client. It does not support extensions or client auth providers.
+func loadClient(kubeconfigPath string) (*k8s.Client, error) {
+	data, err := ioutil.ReadFile(kubeconfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("read kubeconfig: %v", err)
+	}
+
+	// Unmarshal YAML into a Kubernetes config object.
+	var config k8s.Config
+	if err := yaml.Unmarshal(data, &config); err != nil {
+		return nil, fmt.Errorf("unmarshal kubeconfig: %v", err)
+	}
+	return k8s.NewClient(&config)
+}
+
+func start(p *Prometheus) error {
+	client, err := k8s.NewInClusterClient()
+	if err != nil {
+		client, err = loadClient(fmt.Sprintf("%v/.kube/config", os.Getenv("HOME")))
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+	type payload struct {
+		eventype string
+		pod      *corev1.Pod
+	}
+	in := make(chan payload)
+	go func() {
+		var pod corev1.Pod
+		watcher, err := client.Watch(context.Background(), "", &pod)
```

Comment:
I think we need logic to reconnect the watch based on this: https://godoc.org/github.com/ericchiang/k8s#Client.Watch